src/ripple/consensus/LedgerTrie.h

```diff
@@ -512,8 +512,7 @@ class LedgerTrie
         Seq diffSeq;
         std::tie(loc, diffSeq) = find(ledger);
 
-        // Cannot erase root
-        if (loc && loc != root.get())
+        if (loc)
```

Comment:
Just so I understand here, you're not allowing `root` to be deleted, you're allowing it to be cleaned up, and the check on line 537 prevents it from being deleted.

Reply:
Bingo.  `root` is always there and always represents the genesis ledger, so the code below can decrement its tip and branch support, but line 537 prevents it from being deleted.

This arguably should've been the implementation from the start, so I appreciate your suggesting the tests.